wp-content/plugins/core/src/Templates/Components/Abstract_Controller.php

```diff
@@ -59,4 +71,96 @@ public static function factory( array $args = [] ) {
 		return tribe_project()->container()->make( static::class, [ 'args' => $args ] );
 	}
 
+	public function render_breadcrumbs(): void {
+		get_template_part(
+			'components/breadcrumbs/breadcrumbs',
+			'page',
+			[
+				Breadcrumbs_Controller::BREADCRUMBS => $this->get_breadcrumbs(),
+			]
+		);
+	}
+
+	/**
+	 * Returns breadcrumbs for the post.
+	 *
+	 * @return array \Tribe\Project\Templates\Models\Breadcrumb[] Breadcrumbs for the post.
+	 */
+	protected function get_breadcrumbs(): array {
```

Comment:
All this breadcrumb data logic, should this be a separate class, or a Trait? Now all controllers have breadcrumb functionality, which we don't need for many components (Like the button component for example). We really only need them are part of top level  (Page) components.